p2p/protocol.py

```diff
@@ -116,6 +122,120 @@ def encode(self, data: _DecodedMsgType) -> Tuple[bytes, bytes]:
         return header, body
 
 
+class BaseRequest(ABC):
+    """
+    Base representation of a *request* to a connected peer which has a matching
+    *response*.
+    """
+    @abstractmethod
+    def validate_response(self, response: Any) -> None:
+        pass
+
+
+class BaseHeaderRequest(BaseRequest):
+    block_number_or_hash: BlockIdentifier
+    max_headers: int
+    skip: int
+    reverse: bool
+
+    @property
+    @abstractmethod
+    def MAX_HEADERS_FETCH(self) -> int:
```

Comment:
I find it weird to have an all-uppercase property. Any reason for that?

Reply:
it's a constant so I mirrored the casing.  I'm fine lower casing it.